### DISCUSSION TITLE
Gitea

### DISCUSSION THREAD

**Discussion by dmoonfire** [catppuccin/catppuccin#1883] [Port Requests]
### Will you make the port?

- [X] Yes, I'm willing to make the port

### How can the requested port be themed?

Gitea has a bunch of formatting rules, mainly with injected CSS. I've been using it for a while and it seems to be fairly usable.

You can see a working example at https://src.mfgames.com/

### Has the requested port been themed already?

Yes, I have an implementation of all four Catppuccin themes at https://src.mfgames.com/dmoonfire/dmoonfire-gitea-theme-catppuccin

You can create an account and switch between them in the settings.

**Comment by justTOBBI** [comment]
I don't quite know whats the difference, but we have a gitea stylus usertheme under catppuccin/codeberg

**Reply by dmoonfire** [reply]
This one is installed on a Gitea instance, not a user style, so it is available for all the users.

**Comment by jolheiser** [comment]
Would you mind if I added this to [awesome-gitea](https://gitea.com/gitea/awesome-gitea#user-content-themes)?

Very nice work! ❤️ 

---

I might use this is a basis to make a template for people who want to make a theme, this is very well laid out.

**Reply by dmoonfire** [reply]
No problem with adding it to the awesome list.

I'm quite happy with the results and I think others would be also. I did have credits from my source already in the README but the biggest differences from the user styles are:

- Separate theme files as Gitea requires for public assets
- Switched to use CSS variables because they are awesome
- Tweaked a couple of the colors because the backgrounds were too bright with mocha and other dark theme

**Comment by sgoudham** [comment]
Does it make sense to somehow take the contents of this repository and the Gitea userstyle from our codeberg repository and turn it into one `catppuccin/gitea` repository? 

I'm unsure how we would manage the relationship between GitHub and Gitea here. Thoughts? @justTOBBI @jolheiser

**Reply by jolheiser** [reply]
I think that would likely work, would need a little fiddling though. The install-able style needs to be separated like they are so that an instance admin can place them in the correct spot, but the userstyle has them all together, so there might need to be a build step similar to how the palette repo commits generated content.

**Comment by nekowinston** [comment]
Closed via #1990 & #1984:

https://github.com/catppuccin/gitea
https://gitea.catppuccin.com
